### ISSUE TITLE
How to detect missing or empty file field and abort, avoiding Upstream timout exception?

### ISSUE THREAD

**Issue by plato-cambrian** [balderdashy/skipper#89]
When I receive a post to `/api/files.upload`, I need to 
1) verify there is a file that was posted under the name `file` or send 400 response
2) proceed to detect MIME type from magic bytes, send 400 response if non-image
3) upload to s3 with skipper-s3

It's unclear to me how to handle step 1. I've pasted my attempt below. The 1 second timeout helps avoid a race condition where Skipper tries to process the request before it receives the body. If a post is received without the file, Skipper crashes the server shortly after I `return callback('no file buffer available');`.
```
  sopro.files.handleUploadWrapper = function(opts, callback){
    var req = opts.req;
    if (req.file('file')._readableState.buffer.length === 0) {
      setTimeout(function(){
        if (req.file('file')._readableState.buffer.length === 0) {
          return callback('no file buffer available');
        }
        sopro.files.handleUpload(opts, callback); // read magic bytes then call req.file('file').upload()
      }, 1000)
    } else {
      sopro.files.handleUpload(opts, callback);
    }
  }
```

```
Error: EMAXBUFFER: An Upstream (`NOOP_file`) timed out before it was plugged into a receiver. It was still unused after waiting 4500ms. You can configure this timeout by changing the `maxTimeToBuffer` option.
    at null.<anonymous> (C:\dev\src\sopro\apps\chat\node_modules\skipper\standalone\Upstream\Upstream.js:86:15)
    at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)
```

**Comment by sailsbot** [comment]
Thanks for posting, @plato-cambrian.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
